tenant/src/main/scala/com/improving/app/tenant/domain/util.scala

```diff
@@ -3,16 +3,10 @@ package com.improving.app.tenant.domain
 import com.improving.app.common.domain.util.{addressFromEditableAddress, contactFromEditableContact}
 
 object util {
-  def infoFromEditableInfo(infoToUpdate: EditableTenantInfo, infoToCopy: TenantInfo): TenantInfo =
-    infoToCopy
-      .copy(
-        name = infoToUpdate.name.getOrElse(infoToCopy.name),
-        address = infoToUpdate.address
-          .map(addressFromEditableAddress(_, infoToCopy.address))
-          .getOrElse(infoToCopy.address),
-        primaryContact = infoToUpdate.primaryContact
-          .map(contactFromEditableContact(_, infoToCopy.primaryContact))
-          .getOrElse(infoToCopy.primaryContact),
-        organizations = infoToUpdate.organizations.getOrElse(infoToCopy.organizations)
-      )
+  def infoFromEditableInfo(editable: EditableTenantInfo): TenantInfo = TenantInfo(
```

Comment:
Best to incorporate validation here and return an Either to avoid exceptions

Reply:
This function is only used in transition from Draft to Active or to Suspended state in event handler, after command handler has done validation. So I don't think Either is necessary - we will have already validated that these fields exist

Reply:
My previous comment was intended to point out that we should never have validation errors here. If we do, the coder is doing validation wrong in the transition out of Draft state.

Reply:
Right. my suggestion is to change it so we (get closer to) that being a guarantee that is checked at compile time.

Reply:
I don't agree that that is a better state of affairs. The separate between command and event handling can lead to coder error. Checking this at compile time is a red herring to me here - it's more automatic and so seems better. But in reality it is redundant & could lead to confusion about command and event handling separation of concerns (ie only update state in event handling)